src/averageAges.js

```diff
@@ -8,18 +8,25 @@
  *
  * To calculate century:
  * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
+ *  // learn how to use array methods like .filter
+ * .map .some .every .find .reduce
+  // avoid using loop and forEach
+  // replace `if ()` statement with &&, || or ?:
+  // without nesting
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const menArr = people.filter(person => century
+    ? person.sex === 'm' && Math.ceil(person.died / 100) === century
+    : person.sex === 'm');
+
+  const averageMen = menArr.reduce((sum, man) =>
+    sum + (man.died - man.born), 0);
```

Comment:
You can move it to separate function and reuse it in `calculateWomenAverageAge` function

Reply:
ok, thank you!!